README.md
# Vanguard Estates Player

A mini-site that helps folks easily listent to the wonderful [Vanguards Estates choose your own path story](https://www.flashforwardpod.com/2022/10/31/welcome-to-vanguard-estates/) from [Flash Forward](https://www.flashforwardpod.com/).

## Start listening

https://vanguard-estates.wrightkuhns.com

[![Welcome to Vanguard Estates logo](header.jpg)](https://vanguard-estates.wrightkuhns.com)

This is an un-official fan project that I made in hopes that even more people will get to enjoy the story! Fortunately Rose liked it too, so now it's also hosted [on the official Flash Forward website, too!](https://www.flashforwardpod.com/vanguard-interactive/)

## Story structure

A visualization of the paths available in the story

```mermaid
graph LR
    1[Welcome to Vanguard Estates]
    1 --> 2
    1 --> 3
    2[Stay at Vanguard]
    2 --> 6
    2 --> 7
    3[Bring Him Home]
    3 --> 4
    3 --> 5
    4[Buy A Home Robot]
    4 --> 8
    4 --> 9
    5[Call Him More]
    5 --> 10
    5 --> 11
    6[Use Drop-In]
    6 --> 12
    6 --> 13
    7[Disable Drop-In]
    7 --> 14
    7 --> 15
    8[Explain Missy]
    8 --> 16
    8 --> 17
    9[Don't Explain Missy] %% end
    10[Install Cameras]
    10 --> 18
    10 --> 19
    11[Move Him In]
    11 --> 20
    11 --> 21
    12[Don't Tell Him]
    12 --> 22
    12 --> 23
    13[Tell Him]
    13 --> 24
    13 --> 25
    14[Robot Baths]
    14 --> 26
    14 --> 27
    15[Human Baths] %% end
    16[Go To Manila] %% end
    17[Don't Go To Manila] %% end
    18[Handle it Yourself] %% end
    19[Pay for Help] %% end
    20[Drop Out of School] %% end
    21[Take Out a Loan] %% end
    22[Stop the Robot] %% end
    23[Trust the Robot] %% end
    24[Keep His Secret] %% end
    25[Report Him] %% end
    26[Break the News] %% end
    27[Keep Quiet] %% end
```

## Local development

If you've got most-any version of node installed, you can serve this app locally using the following:

```bash
npx serve@latest -l 3027
```


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).

Note, there are two types of releases:
* "Official" – Releases that are available at https://www.flashforwardpod.com/vanguard-interactive/
* "Mirror" - Releases that are available at https://vanguard-estates.wrightkuhns.com

## Mirror 1.1.0 - 2023-09-16

### New

- Initial release at https://vanguard-estates.wrightkuhns.com since the SSL certificate of www.flashforwardpod.com is currently expired.
- Add the `header.jpg` image to the repo.
- Update README URLs to point at the working mirror.
## [Official 1.1.0] - 2022-12-01

### Added

- Added transcripts for all stories!

### Fixed

- Fixed bug where manually scrolling in Firefox for iOS (and perhaps other mobile browser) didn't work well because the expanding & contracting address bar changed the height of each story, causing stories to jump around.
## [Official 1.0.1] - 2022-11-30

### Changed

- Changed welcome/header styling to be horizontally centered on wide screens.

### Fixed

- Fixed the "back" link on the 1st story to scroll the top of the page.
- Fixed keyboard navigation to the "start" link in Firefox for MacOS.
- Fixed welcome/header styling to not be partially covered by displays with notches.

## Official 1.0.0 - 2022-11-29

### Added

- Initial release now live at https://www.flashforwardpod.com/vanguard-interactive/ 🎉
- Ability to easily navigate between all stories & paths.
- Full keyboard navigation.
- Reactive design that works on mobile and desktop devices of all sizes.
- Bonus: Use different story layout modes with the `?tree` or `?single` query params.


[Official 1.1.0]: https://github.com/metavida/vanguard-estates/releases/tag/official-1.1.0
[Official 1.0.1]: https://github.com/metavida/vanguard-estates/releases/tag/official-1.0.1


main.js
/*
Copyright (C) 2022 Marcos Wright-Kuhns
All Rights Reserved
*/
import { storyLayers, allStories } from "./story.js";

let autoPlay;
const setAutoPlay = (value) => {
  autoPlay = !!value;
  window.localStorage.setItem("auto_play", value ? "true" : "");
};

const initAutoPlay = () => {
  autoPlay = window.localStorage.getItem("auto_play");
  const autoPlayEl = document.getElementById("auto_play");
  if (autoPlay === null || autoPlay) {
    setAutoPlay(true);
    autoPlayEl.checked = true;
  } else {
    setAutoPlay(false);
    autoPlayEl.checked = false;
  }
  autoPlayEl.addEventListener("change", ({ target }) =>
    setAutoPlay(target.checked)
  );
};

if ("content" in document.createElement("template")) {
  console.log("A supported browser!");
} else {
  console.warn("Need to polyfill template cloning?");
}

const render = () => {
  const getStartedLink = document.querySelector("header .start > div");
  getStartedLink.addEventListener("click", (event) =>
    focusStory({ storyId: 1, event })
  );
  getStartedLink.addEventListener("keydown", (event) => {
    const { code } = event;
    if (code === "Enter" || code === "Space") {
      focusStory({ storyId: 1, event });
    }
  });
  const getAboutLink = document.querySelector("footer a");
  getAboutLink.addEventListener("click", (event) => focusAbout({ event }));

  if (window.location.search === "?tree") {
    renderTree();
  } else if (window.location.search === "?single") {
    renderVertical({ className: "single" });
  } else {
    renderVertical({ className: "vertical" });
  }
};

const renderVertical = ({ className }) => {
  document.querySelector("html").classList.add(className);
  const main = document.querySelector("main");
  main.classList.add(className);

  for (let id = 1; id < Object.keys(allStories).length; id++) {
    const story = allStories[id];
    renderStory({ story, parentEl: main });
  }
};

const renderTree = () => {
  document.querySelector("html").classList.add("tree");
  const main = document.querySelector("main");
  main.classList.add("tree");

  storyLayers.forEach((basicStories, index) => {
    const stories = basicStories.map(({ id }) => allStories[id]);
    renderRow({
      stories,
      parentEl: main,
      index,
    });
  });
};

const renderRow = ({ stories, parentEl, index }) => {
  const rowEl = document.createElement("div");
  rowEl.setAttribute("id", `row-${index}`);
  rowEl.classList.add("row");
  stories.forEach((story) => {
    renderStory({ story, parentEl: rowEl });
  });

  parentEl.append(rowEl);
};

const renderStory = ({ story, parentEl }) => {
  const storyEl = document.createElement("div");
  storyEl.classList.add("story");
  storyEl.setAttribute("tabindex", "-1");

  if (story.id) {
    const { id: storyId, name, children, parent } = story;
    storyEl.setAttribute("id", `story-${storyId}`);

    const cellTemp = document.getElementById("cell_template");
    storyEl.append(cellTemp.content.cloneNode(true));

    storyEl.querySelector("h2").textContent = `${storyId}. ${name}`;
    const imgEl = storyEl.querySelector("img");
    imgEl.setAttribute(
      "src",
      `https://megaphone.imgix.net/podcasts/${story.image.megaphone}?max-w=300&max-h=300&fit=crop&auto=format,compress`
    );
    imgEl.setAttribute("alt", story.image.alt);

    const audioEl = storyEl.querySelector(".player audio");
    audioEl.setAttribute("src", story.audio.url);
    audioEl.addEventListener("play", () =>
      handlePlayStart({ audioEl, storyEl, storyId })
    );
    audioEl.addEventListener("pause", () => pauseCurrentAudio({ audioEl }));

    const transcriptEl = storyEl.querySelector("details");
    loadTranscript({ storyEl, storyId });
    const transcriptToggleEl = transcriptEl.querySelector("summary");
    transcriptToggleEl.addEventListener("click", (event) => {
      toggleTranscript({ storyEl });
    });
    transcriptToggleEl.addEventListener("keydown", (event) => {
      const { code } = event;
      if (code === "Enter" || code === "Space") {
        event.preventDefault();
        toggleTranscript({ storyEl });
      }
    });

    const childrenEl = storyEl.querySelector(".children");
    const childrenList = childrenEl.querySelector("ol");
    if (children.length) {
      children.forEach((childId) =>
        renderStoryNavLink({
          storyId: childId,
          parentEl: childrenList,
          pauseCurrent: true,
        })
      );
    } else {
      const endEl = childrenEl.querySelector("p");
      endEl.textContent = "The End";

      renderStoryNavLink({
        storyId: 1,
        parentEl: childrenList,
        className: "restart",
        pauseCurrent: true,
      });
    }
    if (parent) {
      renderStoryNavLink({
        storyId: parent,
        parentEl: childrenList,
        className: "back",
        noPlay: true,
        pauseCurrent: true,
      });
    } else {
      renderAboutNavLink({
        parentEl: childrenList,
      });
    }

    storyEl.addEventListener("click", (event) => {
      focusStory({
        storyId,
        event,
        // Don't try to auto-play if we clicked inside the transcript area.
        noPlay: event.composedPath().includes(transcriptEl),
      });
    });
    storyEl.addEventListener("keydown", (event) => {
      const { target, code } = event;
      if (target === storyEl) {
        if (code === "Enter" || code === "Space") {
          event.preventDefault();
          togglePlayPause({ audioEl });
        }
      }
    });
  } else {
    storyEl.classList.add("hidden");
  }

  parentEl.append(storyEl);
};

const togglePlayPause = ({ audioEl }) => {
  if (audioEl.classList.contains("playing")) {
    audioEl.pause();
  } else {
    audioEl.play();
  }
};
const pauseCurrentAudio = () => {
  const prevAudioEl = document.querySelector(".playing");
  if (prevAudioEl) {
    prevAudioEl.classList.remove("playing");
    prevAudioEl.pause();
  }
};
const handlePlayStart = ({ audioEl, storyEl, storyId }) => {
  pauseCurrentAudio();
  audioEl.classList.add("playing");
  if (!storyEl.classList.contains("focused")) {
    focusStory({ storyId, noPlay: true });
  }
};

const toggleTranscript = ({ storyEl }) => {
  const transcriptEl = storyEl.querySelector("details");
  if (transcriptEl.open) {
    storyEl.classList.remove("show-transcript");
    transcriptEl.open = false;
  } else {
    storyEl.classList.add("show-transcript");
    transcriptEl.open = true;
  }
};

const loadTranscript = async ({ storyEl, storyId }) => {
  const transcriptHolderEl = storyEl.querySelector(".transcript");
  if (transcriptHolderEl.dataset.fetched === "false") {
    transcriptHolderEl.dataset.fetched = "true";
    const transcriptText = await fetchTranscript({ storyId });
    transcriptHolderEl.innerHTML = transcriptText;
  }
};

const fetchTranscript = async ({ storyId }) => {
  try {
    const resp = await fetch(`./transcripts/${storyId}.html`);
    if (!resp.ok) {
      throw "Failed to fetch story";
    }
    const body = await resp.text();
    return body;
  } catch (err) {
    return `Failed to load transcript for Story ${storyId}.`;
  }
};

const renderNavLink = ({ parentEl, number, text, url, onClick, className }) => {
  const listEl = document.createElement("li");
  listEl.setAttribute("value", number);
  listEl.classList.add(className || "child");
  listEl.setAttribute("tabindex", "0");
  listEl.addEventListener("click", onClick);
  listEl.addEventListener("keydown", (event) => {
    const { code } = event;
    if (code === "Enter" || code === "Space") {
      event.preventDefault();
      onClick();
    }
  });

  const storyLink = document.createElement("a");
  storyLink.setAttribute("href", url);
  storyLink.setAttribute("tabindex", "-1");

  const linkText = document.createElement("span");
  linkText.textContent = text;

  storyLink.append(linkText);
  listEl.append(storyLink);
  parentEl.append(listEl);
};
const renderStoryNavLink = ({
  storyId,
  parentEl,
  className,
  noPlay,
  pauseCurrent,
}) => {
  renderNavLink({
    parentEl,
    number: storyId,
    text: allStories[storyId].name,
    className,
    url: `#view-story-${storyId}`,
    onClick: (event) =>
      focusStory({ storyId: storyId, event, noPlay, pauseCurrent }),
  });
};
const renderAboutNavLink = ({ parentEl }) => {
  renderNavLink({
    parentEl,
    number: 1,
    text: "About this site",
    className: "back",
    url: "#welcome",
    onClick: (event) => focusAbout({ event }),
  });
};

const initialFocus = () => {
  const [_, storyId] = window.location.hash?.match(/#view-story-(\d+)/) || [];
  if (storyId) {
    focusStory({ storyId, noHistory: true });
  } else if (window.location.hash === "#welcome") {
    focusAbout({ noHistory: true });
  } else {
    window.scrollTo(0, 0);
  }
};

const setTitle = (...extraText) => {
  document.title = [
    ...extraText,
    "Welcome to Vanguard Estates",
    "Fash Forward",
  ].join(" | ");
};

const scrollAndUpdateHistory = ({ event, targetEl, history }) => {
  event?.preventDefault();

  targetEl.scrollIntoView({
    behavior: "smooth",
    block: "center",
    inline: "center",
  });

  if (history) {
    const { state, url } = history;
    window.history.pushState(state, "", url);
  }
};

const focusStory = ({ storyId, event, noHistory, noPlay, pauseCurrent }) => {
  document.querySelector(".focused")?.classList.remove("focused");
  event && event.stopPropagation();
  if (pauseCurrent) {
    pauseCurrentAudio();
  }

  const storyEl = document.getElementById(`story-${storyId}`);

  const audioEl = storyEl.querySelector(".player audio");
  if (event && event.target !== audioEl && autoPlay && !noPlay) {
    audioEl.play();
  }

  storyEl.classList.add("focused");
  document.querySelector("footer").classList.remove("hidden");

  storyEl.focus();
  scrollAndUpdateHistory({
    event,
    targetEl: storyEl,
    history: noHistory
      ? null
      : { state: { storyId }, url: `#view-story-${storyId}` },
  });
  setTitle(allStories[storyId]?.name);
};

const focusAbout = ({ event, noHistory }) => {
  document.querySelector("footer").classList.add("hidden");
  event && event.stopPropagation();
  document.querySelector("header .start > div").focus();
  scrollAndUpdateHistory({
    event,
    targetEl: document.querySelector("header > img"),
    history: noHistory ? null : { url: "#welcome" },
  });
  setTitle();
};

document.addEventListener("DOMContentLoaded", () => {
  render();
  initAutoPlay();
  initialFocus();
  window.addEventListener("popstate", initialFocus);
});


history.js
import { allStories } from "./story.js";

const initStoryHistory = (story) => {
  let numChildren = 0;
  if(story.children.length) {
    numChildren += story.children.length;
    story.children.forEach(storyId => {
      numChildren += initStoryHistory(allStories[storyId]);
    })
  }
  // TODO: start numListened at zero
  const numListened = Math.round(numChildren * Math.random());
  const history = (
    JSON.parse(window.localStorage.getItem(`story-${story.id}`))
    || { numListened: numListened, isListened: false }
  );
  // TODO: collect all parents
  Object.assign(allStories[story.id], history);
  return numChildren;
};
initStoryHistory(allStories[1]);

// Expose a set of APIs so that the UI
// can locally track which stories have been listened to.

// Given a storyId
// returns an object shaped like this:
// { isListenend: <bool>, numChildren: <int>, numListened: <int> }
const historyForStory = ({ storyId }) => {
  const { numChildren, numListened } = allStories[storyId]

  return {
    isListened: false,
    numChildren,
    numListened,
  };
};

export const setNavLinkHistory = ({ historyEl, storyId }) => {
  const checkImg = historyEl.querySelector("svg");
  const { numChildren, numListened } = historyForStory({ storyId })
  const percentListened = numListened / numChildren;
  if (numListened === 0) {
    historyEl.setAttribute("data-listened", "0");
  } else if (percentListened <= 0.25) {
    historyEl.setAttribute("data-listened", "25");
  } else if (percentListened <= 0.5) {
    historyEl.setAttribute("data-listened", "50");
  } else if (percentListened < 1) {
    historyEl.setAttribute("data-listened", "75");
  }
  checkImg.setAttribute("alt", `(you've listened to ${numListened} of ${numChildren} stories on this path)`);
  historyEl.setAttribute("title", `(you've listened to ${numListened} of ${numChildren} stories on this path)`);
}

export const markStoryListened = ({ storyId }) => {
  // TODO: Increment numListened for all parents
};


story.js
/*
Copyright (C) 2022 Flash Forward (https://www.flashforwardpod.com)
All Rights Reserved

All data in this file originally comes from the FlashForward RSS feed:
https://feeds.megaphone.fm/PPY8359579005
To see the original visit:
https://www.flashforwardpod.com/2022/10/31/welcome-to-vanguard-estates/
*/

export const storyLayers = [
  [{ id: "1", name: "Welcome to Vanguard Estates", children: [2, 3] }],
  [
    { id: "2", name: "Stay at Vanguard", children: [6, 7], parent: 1 },
    { id: "3", name: "Bring Him Home", children: [4, 5], parent: 1 },
  ],
  [
    { id: "6", name: "Use Drop-In", children: [12, 13], parent: 2 },
    { id: "7", name: "Disable Drop-In", children: [14, 15], parent: 2 },
    { id: "4", name: "Buy A Home Robot", children: [8, 9], parent: 3 },
    { id: "5", name: "Call Him More", children: [10, 11], parent: 3 },
  ],
  [
    { id: "12", name: "Don't Tell Him", children: [22, 23], parent: 6 },
    { id: "13", name: "Tell Him", children: [24, 25], parent: 6 },
    { id: "14", name: "Robot Baths", children: [26, 27], parent: 7 },
    { id: "15", name: "Human Baths", children: [], parent: 7 },
    { id: "8", name: "Explain Missy", children: [16, 17], parent: 4 },
    { id: "9", name: "Don't Explain Missy", children: [], parent: 4 },
    { id: "10", name: "Install Cameras", children: [18, 19], parent: 5 },
    { id: "11", name: "Move Him In", children: [20, 21], parent: 5 },
  ],
  [
    { id: "22", name: "Stop the Robot", children: [], parent: 12 },
    { id: "23", name: "Trust the Robot", children: [], parent: 12 },
    { id: "24", name: "Keep His Secret", children: [], parent: 13 },
    { id: "25", name: "Report Him", children: [], parent: 13 },
    { id: "26", name: "Break the News", children: [], parent: 14 },
    { id: "27", name: "Keep Quiet", children: [], parent: 14 },
    {},
    {},
    { id: "16", name: "Go To Manila", children: [], parent: 8 },
    { id: "17", name: "Don't Go To Manila", children: [], parent: 8 },
    {},
    {},
    { id: "18", name: "Handle it Yourself", children: [], parent: 10 },
    { id: "19", name: "Pay for Help", children: [], parent: 10 },
    { id: "20", name: "Drop Out of School", children: [], parent: 11 },
    { id: "21", name: "Take Out a Loan", children: [], parent: 11 },
  ],
];

// Note: I couldn't find original alt text so I wrote this up myself.
const storyImages = {
  1: {
    orig: "originals/art/01.png",
    megaphone: "bc5a644a-531b-11ed-a193-3b3da2ee79f8/image/3724cf.png",
    alt: "A robot with butler outfit carrying covered a tray of food",
  },
  2: {
    orig: "originals/art/02.png",
    megaphone: "4d6eaec2-54cb-11ed-9989-73fd73486a64/image/80982b.png",
    alt: "You, looking at a laptop screen",
  },
  3: { orig: "originals/art/03.png", megaphone: "d09fe8ba-54cb-11ed-ab3f-17c41edd528f/image/99f655.png", alt: "A frowning robot" },
  4: {
    orig: "originals/art/04.png",
    megaphone: "345aee38-54cf-11ed-9cdc-df092e488f0e/image/3d4fb1.png",
    alt: "A cute, orange cat robot",
  },
  5: {
    orig: "originals/art/05.png",
    megaphone: "3bdde678-54d0-11ed-974c-97d5c6e2ae87/image/cccea9.png",
    alt: "You, on the phone with your dad with your palm on your face",
  },
  6: {
    orig: "originals/art/06.png",
    megaphone: "468584c8-54d5-11ed-942c-ab6c5f0c087a/image/5fb8be.png",
    alt: "You, dumping a bag of rice on an open laptop",
  },
  7: {
    orig: "originals/art/07.png",
    megaphone: "51351992-54d5-11ed-88a5-676e836ac2c0/image/d701fa.png",
    alt: "A group of robots staring at your, one with police lights on top of it",
  },
  8: {
    orig: "originals/art/08.png",
    megaphone: "6167320a-54d5-11ed-b771-ff625a1e688a/image/3721d6.png",
    alt: "A cat robot looking happily down at a cell phone, about to place a call",
  },
  9: {
    orig: "originals/art/09.png",
    megaphone: "6ce60214-54d5-11ed-bc92-3f49041d982d/image/e07435.png",
    alt: "Your dad with a huge cat eyeball staring at him",
  },
  10: {
    orig: "originals/art/10.png",
    megaphone: "759370ea-54d5-11ed-98d2-2bad4aa9c368/image/307f03.png",
    alt: "Skid marks on a road lead to a fire and a tire bouncing away from the fire",
  },
  11: {
    orig: "originals/art/11.png",
    megaphone: "800375d4-54d5-11ed-afb0-7bc7a6304ef9/image/a8eb54.png",
    alt: "A house that appears to be bursting at the seems with two huge people visible through the window, filling the whole house",
  },
  12: {
    orig: "originals/art/12.png",
    megaphone: "8cc25ae2-54d5-11ed-a338-ff7c7799f749/image/efa9b2.png",
    alt: "A smirking robot wearing a trench coat that is concealing pill bottels",
  },
  13: {
    orig: "originals/art/13.png",
    megaphone: "97f131b8-54d5-11ed-b560-9366327710f1/image/63e7ed.png",
    alt: "A huge eyeball staring down at your dad who is cowering on the floor",
  },
  14: {
    orig: "originals/art/14.png",
    megaphone: "a37779d4-54d5-11ed-a45b-ff3a1a7e8755/image/c6486e.png",
    alt: "Your dad, barely visible because he's covered in bubbles and being scrubbed by huge revolving brushes",
  },
  15: {
    orig: "originals/art/15.png",
    megaphone: "ad56697e-54d5-11ed-823a-efe0254447bb/image/2354d7.png",
    alt: "You dad, peeking out of a bathtub, looking nervous",
  },
  16: {
    orig: "originals/art/16.png",
    megaphone: "c2b6c250-54d5-11ed-8743-53c00e0a0efb/image/a470ff.png",
    alt: "An airplane, flying high above the clouds",
  },
  17: {
    orig: "originals/art/17.png",
    megaphone: "d59f2f7e-54d5-11ed-a45b-d787246b22a2/image/784cd2.png",
    alt: "A robot cat, dead and laying on its side",
  },
  18: {
    orig: "originals/art/18.png",
    megaphone: "df58ca7a-54d5-11ed-a9a2-bf14d50a6098/image/ab074a.png",
    alt: "You, handing your dad his car keys",
  },
  19: {
    orig: "originals/art/19.png",
    megaphone: "e8ca7d4c-54d5-11ed-8093-03154630b9cb/image/a7a236.png",
    alt: "A broken piggy bank",
  },
  20: {
    orig: "originals/art/20.png",
    megaphone: "f32b1882-54d5-11ed-8790-ab1837edb444/image/74a878.png",
    alt: "A graduation hat with a lit fuse instead of a tassle",
  },
  21: {
    orig: "originals/art/21.png",
    megaphone: "fe6f3200-54d5-11ed-a07e-ab476799fa89/image/d01e48.png",
    alt: "You are wearing handcuffs while a person hands you money",
  },
  22: {
    orig: "originals/art/22.png",
    megaphone: "06c2527a-54d6-11ed-833b-9bac029a2b27/image/f46b0a.png",
    alt: "You, speaking into a microphone",
  },
  23: {
    orig: "originals/art/23.png",
    megaphone: "0ffb2a74-54d6-11ed-83df-170123454644/image/378d52.png",
    alt: "Your dad, asleep in a chair",
  },
  24: {
    orig: "originals/art/24.png",
    megaphone: "1b807926-54d6-11ed-9ff3-ab199d638f1f/image/d3060c.png",
    alt: "Two people holding hands",
  },
  25: {
    orig: "originals/art/25.png",
    megaphone: "24f74610-54d6-11ed-b7aa-63153bdcac29/image/aef4b6.png",
    alt: "Two people on opposite sides of a brick wall",
  },
  26: {
    orig: "originals/art/26.png",
    megaphone: "2efdf65e-54d6-11ed-be95-9b0bd23baa80/image/387a30.png",
    alt: "Your dad, covering his face while three robot hands reach out & covering his head",
  },
  27: {
    orig: "originals/art/27.png",
    megaphone: "3983a380-54d6-11ed-9a65-a3b7565ec3b2/image/9a35d2.png",
    alt: "Your dad in his bathtub with huge, stadium lights shining down on him",
  },
};

// Note: duration comes from <itunes:duration> which Apple says
// should be episode length in seconds:
// https://help.apple.com/itc/podcasts_connect/#/itcb54353390
const storyAudio = {
  1: {
    url: "https://www.podtrac.com/pts/redirect.mp3/traffic.megaphone.fm/FLASH8662401899.mp3?updated=1667061610",
    duration: 515,
  },
  2: {
    url: "https://www.podtrac.com/pts/redirect.mp3/traffic.megaphone.fm/FLASH8906426865.mp3?updated=1667062002",
    duration: 187,
  },
  3: {
    url: "https://www.podtrac.com/pts/redirect.mp3/traffic.megaphone.fm/FLASH1879514191.mp3?updated=1667062298",
    duration: 214,
  },
  4: {
    url: "https://www.podtrac.com/pts/redirect.mp3/traffic.megaphone.fm/FLASH4091170562.mp3?updated=1667062541",
    duration: 188,
  },
  5: {
    url: "https://www.podtrac.com/pts/redirect.mp3/traffic.megaphone.fm/FLASH4076018928.mp3?updated=1667062595",
    duration: 245,
  },
  6: {
    url: "https://www.podtrac.com/pts/redirect.mp3/traffic.megaphone.fm/FLASH6513354357.mp3?updated=1667063031",
    duration: 184,
  },
  7: {
    url: "https://www.podtrac.com/pts/redirect.mp3/traffic.megaphone.fm/FLASH5443970620.mp3?updated=1667063497",
    duration: 281,
  },
  8: {
    url: "https://www.podtrac.com/pts/redirect.mp3/traffic.megaphone.fm/FLASH9190312190.mp3?updated=1667063567",
    duration: 327,
  },
  9: {
    url: "https://www.podtrac.com/pts/redirect.mp3/traffic.megaphone.fm/FLASH2709325782.mp3?updated=1667063917",
    duration: 292,
  },
  10: {
    url: "https://www.podtrac.com/pts/redirect.mp3/traffic.megaphone.fm/FLASH2773947020.mp3?updated=1667064293",
    duration: 229,
  },
  11: {
    url: "https://www.podtrac.com/pts/redirect.mp3/traffic.megaphone.fm/FLASH2197377291.mp3?updated=1667064743",
    duration: 302,
  },
  12: {
    url: "https://www.podtrac.com/pts/redirect.mp3/traffic.megaphone.fm/FLASH5695864686.mp3?updated=1667064938",
    duration: 162,
  },
  13: {
    url: "https://www.podtrac.com/pts/redirect.mp3/traffic.megaphone.fm/FLASH2788464413.mp3?updated=1667065257",
    duration: 367,
  },
  14: {
    url: "https://www.podtrac.com/pts/redirect.mp3/traffic.megaphone.fm/FLASH5338361822.mp3?updated=1667065677",
    duration: 319,
  },
  15: {
    url: "https://www.podtrac.com/pts/redirect.mp3/traffic.megaphone.fm/FLASH8509368617.mp3?updated=1667065801",
    duration: 260,
  },
  16: {
    url: "https://www.podtrac.com/pts/redirect.mp3/traffic.megaphone.fm/FLASH4767887889.mp3?updated=1667066160",
    duration: 242,
  },
  17: {
    url: "https://www.podtrac.com/pts/redirect.mp3/traffic.megaphone.fm/FLASH5505988844.mp3?updated=1667066417",
    duration: 250,
  },
  18: {
    url: "https://www.podtrac.com/pts/redirect.mp3/traffic.megaphone.fm/FLASH4468228646.mp3?updated=1667066603",
    duration: 188,
  },
  19: {
    url: "https://www.podtrac.com/pts/redirect.mp3/traffic.megaphone.fm/FLASH9562689317.mp3?updated=1667066821",
    duration: 181,
  },
  20: {
    url: "https://www.podtrac.com/pts/redirect.mp3/traffic.megaphone.fm/FLASH2307594154.mp3?updated=1667067048",
    duration: 343,
  },
  21: {
    url: "https://www.podtrac.com/pts/redirect.mp3/traffic.megaphone.fm/FLASH7386444037.mp3?updated=1667067214",
    duration: 330,
  },
  22: {
    url: "https://www.podtrac.com/pts/redirect.mp3/traffic.megaphone.fm/FLASH3694163619.mp3?updated=1667067528",
    duration: 296,
  },
  23: {
    url: "https://www.podtrac.com/pts/redirect.mp3/traffic.megaphone.fm/FLASH4864386271.mp3?updated=1667067633",
    duration: 392,
  },
  24: {
    url: "https://www.podtrac.com/pts/redirect.mp3/traffic.megaphone.fm/FLASH5073936630.mp3?updated=1667067886",
    duration: 316,
  },
  25: {
    url: "https://www.podtrac.com/pts/redirect.mp3/traffic.megaphone.fm/FLASH2850365548.mp3?updated=1667067888",
    duration: 282,
  },
  26: {
    url: "https://www.podtrac.com/pts/redirect.mp3/traffic.megaphone.fm/FLASH4294637277.mp3?updated=1667067909",
    duration: 156,
  },
  27: {
    url: "https://www.podtrac.com/pts/redirect.mp3/traffic.megaphone.fm/FLASH5607806547.mp3?updated=1667067952",
    duration: 130,
  },
};

export const allStories = storyLayers.reduce((acc, stories) => {
  const newStories = stories.reduce((acc, story) => {
    acc[story.id] = {
      image: storyImages[story.id],
      audio: storyAudio[story.id],
      ...story,
    };
    return acc;
  }, {});
  Object.assign(acc, newStories);
  return acc;
}, {});
